FrontEnd/src/components/SignUp/components/signup/signup-form/SignUpFormContent.js

```diff
@@ -68,12 +73,32 @@ export function SignUpFormContentComponent(props) {
     const otherOption = name === 'yurosoba' ? 'fop' : 'yurosoba';
     setValue(otherOption, false);
     if (!getValues('yurosoba') && !getValues('fop')) {
-      setError('businessEntity', { type: 'manual', message: errorMessageTemplates.required });
+      setError('businessEntity', { type: 'manual', message: errorMessageTemplates.requiredRepresentative });
     } else {
       clearErrors('businessEntity');
     }
   };
 
+  const validateNameSurname = (value) => {
+    const allowedSymbolsPattern = /^[a-zA-Zа-щюяьА-ЩЮЯЬїЇіІєЄґҐ'\s]+$/;
+    const letterCount = (value.match(/[a-zA-Zа-щюяьА-ЩЮЯЬїЇіІєЄґҐ]/g) || []).length;
+    if (!allowedSymbolsPattern.test(value)) {
+      return errorMessageTemplates.notAllowedSymbols;
+    }
+    if (letterCount < 2) {
+      return errorMessageTemplates.nameSurnameFieldLength;
+    }
+    return true;
+  };
+
+  const onBlurHandler = (fieldName) => {
+    let fieldValue = getValues(fieldName);
```

Comment:
It's not clear what happens if `getValues(fieldName)` returns an undefined or null value. It would be good to handle such cases to avoid calling methods on undefined

Reply:
Check added in if statement.